chaincode/marbles02_private/go/marbles_chaincode_private.go

```diff
@@ -649,3 +655,29 @@ func getQueryResultForQueryString(stub shim.ChaincodeStubInterface, queryString
 
 	return buffer.Bytes(), nil
 }
+
+// ===============================================
+// getMarbleHash - use the public data hash to verify a private marble
+// Result is the hash on the public ledger of a marble stored a private data collection
+// ===============================================
+func (t *SimpleChaincode) getMarbleHash(stub shim.ChaincodeStubInterface, args []string) pb.Response {
```

Comment:
For verification in the secured asset transfer sample, I was taking the approach of passing in the preimage, and then in the chaincode taking the sha256 hash of it, comparing it to the hash returned by GetPrivateDataHash, and then returning true or false as verified.

It can work either way... but I was preferring to take the hash in the chaincode, to save clients from this burden, and to show how to take a sha256 hash in chaincode.

Reply:
I agree, you would not use this function in a realistic chaincode. The intent was to put out a sample function with the API so users can work with it, and understand the concept. Even with the basic private data tutorial, you can get the hashes on marbles that you create, see that the hashes match, and demonstrate how you would use the `GetPrivateDataHash` function. 

If we re-write this tutorial to make it more realistic, we would replace this function. However, this would allow uses to get a basic feel for how it works. Whether this is better than nothing would depend on how long you think it would take to update this chaincode, or whether this function would lead users astray.

Reply:
@denyeart FYI

Reply:
yeah i agree, good to have examples of the various private data functions in the main private data sample, even if unrealistic usage.